code/modules/mob/inventory.dm

```diff
@@ -292,8 +292,9 @@
 //for when you want the item to end up on the ground
 //will force move the item to the ground and call the turf's Entered
 /mob/proc/dropItemToGround(obj/item/I, force = FALSE)
-	I.pixel_x = rand(-6,6)
-	I.pixel_y = rand(-6,6)
+	if(I)
+		I.pixel_x = rand(-6,6)
+		I.pixel_y = rand(-6,6)
 	return doUnEquip(I, force, drop_location(), FALSE)

```

Comment:
`doUnEquip` passes TRUE up to any if's if you didn't have an item, and also has checks for no_drop items and such. 

Just doing a quick check through the codebase, there's 196 current calls for dropItemToGround(), of which a GOOD number of them aren't sanity checking simply because you never had to - the old behavior was that you could safely pass an item slot to it to either drop the item, or continue if the item didn't exist, (and for example, then place something in the slot after. )

Changing it to do sanity checks BEFORE each call will require me to duplicate a fuckton of code and put in a bunch of special case crap (including even the act of disarming people in combat) for no reason other than the fact that we now offset the pixels. 

I don't have a couple hours to rewrite pretty much all of these instances and test every individual one, so feel free to close this if you feel that maintaining the old behavior of how `dropItemToGround`  has **ALWAYS** worked isn't sufficient.

Reply:
then the pixel shifts should be moved after the call to doUnequip and it's return value checked

Reply:
oh and the proc should get a documentaiton flag indicating that passing null in is common

Reply:
Didn't think about nodrop items getting offset, good catch.